pkg/controller/disconnected_operator.go

```diff
@@ -0,0 +1,119 @@
+	"github.com/openshift/library-go/pkg/controller/controllercmd"
+
+	"github.com/openshift/insights-operator/pkg/config"
+	"github.com/openshift/insights-operator/pkg/config/configobserver"
+	"github.com/openshift/insights-operator/pkg/controller/periodic"
+	"github.com/openshift/insights-operator/pkg/controller/status"
+	"github.com/openshift/insights-operator/pkg/gather"
+	"github.com/openshift/insights-operator/pkg/gather/clusterconfig"
+	"github.com/openshift/insights-operator/pkg/recorder"
+	"github.com/openshift/insights-operator/pkg/recorder/diskrecorder"
+)
+
+type DisconnectedOperator struct {
+	config.Controller
+}
+
+func (d *DisconnectedOperator) Run(ctx context.Context, controller *controllercmd.ControllerContext) error {
+	klog.Infof("Starting insights-operator %s", version.Get().String())
+	initialDelay := 0 * time.Second
+	cont, err := config.LoadConfig(d.Controller, controller.ComponentConfig.Object, config.ToDisconnectedController)
+	if err != nil {
+		return err
+	}
+	d.Controller = cont
+
+	// these are operator clients
+	kubeClient, err := kubernetes.NewForConfig(controller.ProtoKubeConfig)
+	if err != nil {
+		return err
+	}
+	configClient, err := configv1client.NewForConfig(controller.KubeConfig)
+	if err != nil {
+		return err
+	}
+	// these are gathering clients
+	gatherProtoKubeConfig := rest.CopyConfig(controller.ProtoKubeConfig)
+	if len(d.Impersonate) > 0 {
+		gatherProtoKubeConfig.Impersonate.UserName = d.Impersonate
+	}
+	gatherKubeConfig := rest.CopyConfig(controller.KubeConfig)
+	if len(d.Impersonate) > 0 {
+		gatherKubeConfig.Impersonate.UserName = d.Impersonate
+	}
+
+	// the metrics client will connect to prometheus and scrape a small set of metrics
+	// TODO: the oauth-proxy and delegating authorizer do not support Impersonate-User,
+	//   so we do not impersonate gather
+	metricsGatherKubeConfig := rest.CopyConfig(controller.KubeConfig)
+	metricsGatherKubeConfig.CAFile = "/var/run/configmaps/service-ca-bundle/service-ca.crt"
+	metricsGatherKubeConfig.NegotiatedSerializer = scheme.Codecs
+	metricsGatherKubeConfig.GroupVersion = &schema.GroupVersion{}
+	metricsGatherKubeConfig.APIPath = "/"
+	metricsGatherKubeConfig.Host = "https://prometheus-k8s.openshift-monitoring.svc:9091"
+
+	// If we fail, it's likely due to the service CA not existing yet. Warn and continue,
+	// and when the service-ca is loaded we will be restarted.
+	gatherKubeClient, err := kubernetes.NewForConfig(gatherProtoKubeConfig)
+	if err != nil {
+		return err
+	}
+	// ensure the insight snapshot directory exists
+	if _, err := os.Stat(d.StoragePath); err != nil && os.IsNotExist(err) {
+		if err := os.MkdirAll(d.StoragePath, 0777); err != nil {
+			return fmt.Errorf("can't create --path: %v", err)
+		}
+	}
+
+	// configobserver synthesizes all config into the status reporter controller
+	configObserver := configobserver.New(d.Controller, kubeClient)
+	go configObserver.Start(ctx)
```

Comment:
Do we need this when triggering manually? I think we can read the config once and don't need to periodically check. It's probably good time to start thinking about the way how user can use it and trigger it. If he/she creates a new deployment then we will need this.....but I think it would be easier to recommend it as one time job, but then there's the question where the archive would be stored